patches/api/0419-Player-standing-on-position-API.patch

```diff
@@ -0,0 +1,55 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 9 Jun 2023 22:49:45 -0400
+Subject: [PATCH] Player standing on position API
+
+
+diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
+index a2a423d4e4c2702ba5967223cab0432dd7d04732..6fa7d251f54be9d0ef6b0ec0283efa927bae1a0b 100644
+--- a/src/main/java/org/bukkit/entity/Entity.java
++++ b/src/main/java/org/bukkit/entity/Entity.java
+@@ -954,4 +954,44 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
+      */
+     boolean wouldCollideUsing(@NotNull BoundingBox boundingBox);
+     // Paper End - Collision API
++    // Paper start - Player standing on position API
++    /**
++     * Gets the block that is currently being used to calculate movement affects (i.e friction) on the entity.
++     * <p>
++     * Examples:
++     * - When sanding on a slab under an ice block, this will return the position of the ice block.
++     * - When flying on top of a flower located under a grass block this will return the position of the grass block (although this technically will not affect your movement).
++     *
++     * @return block position used for movement affects
++     */
++    @NotNull
++    io.papermc.paper.math.BlockPosition getBlockPosBelowThatAffectsMyMovement();
```

Comment:
:eyes: This really feels like a spigot method name :tm:

Reply:
Fixed hopefully